compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -360,14 +385,33 @@ bool TaggedShapeAnalyzer::can_remove_transposes()
   assert(_in != nullptr && _front_transpose != nullptr && _mid_reshape != nullptr &&
          _back_transpose != nullptr);
 
-  init_shape_with_tag(_in_shape_v);
-
-  analyze_transpose(_front_perm_v);
+  auto count_not_1_dim = [](std::vector<int32_t> &shape) -> int {
```

Comment:
(Optional)
How about passing by const reference if the parameter does not modified within the lambda.
```suggestion
  auto count_not_1_dim = [](const std::vector<int32_t> &shape) -> int {
```